From 81666bab3e9a01784ad644af61da58ff8a76b637 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 3 Aug 2021 11:32:30 +0100
Subject: [PATCH 530/552] overlays: Reduce Pi 4 vc4 CMA size to 320MB

Reduce the default CMA allocation requested by the vc4-kms-v3d-pi4 and
vc4-fkms-v3d-pi4 overlays to 320MB.

Use magic values of the form (<n>*64 - 4)MB to encode default values
of <n>*64MB, allowing these defaults to be distinguished from values
set explicitly by the user with the usual overlay parameters (e.g.
"cma-384"). Only default values will be capped if the Pi RAM is too
small or the gpu_mem setting too large for it to be viable.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/vc4-fkms-v3d-pi4-overlay.dts | 2 +-
 arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/vc4-fkms-v3d-pi4-overlay.dts b/arch/arm/boot/dts/overlays/vc4-fkms-v3d-pi4-overlay.dts
index bb6731e1c9c4..7792ead0cbb3 100644
--- a/arch/arm/boot/dts/overlays/vc4-fkms-v3d-pi4-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-fkms-v3d-pi4-overlay.dts
@@ -8,7 +8,7 @@
 #include "cma-overlay.dts"
 
 &frag0 {
-       size = <(512*1024*1024)>;
+	size = <((320-4)*1024*1024)>;
 };
 
 / {
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts b/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
index 0480fa7a8129..4285e12a4e53 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
@@ -10,7 +10,7 @@
 #include "cma-overlay.dts"
 
 &frag0 {
-       size = <(512*1024*1024)>;
+	size = <((320-4)*1024*1024)>;
 };
 
 / {
-- 
2.33.0

